mbed_tools_ci/generate_docs.py

```diff
@@ -2,27 +2,45 @@
 import argparse
 import logging
 import sys
+import shutil
 from subprocess import check_call
+from pathlib import Path
 
-from mbed_tools_ci.utils.configuration import configuration, ConfigurationVariable
+from mbed_tools_ci.utils.configuration import (
+    configuration,
+    ConfigurationVariable,
+)
 from mbed_tools_ci.utils.logging import log_exception
 
 logger = logging.getLogger(__name__)
 
 
-def generate_docs(output_directory: str) -> int:
-    """Triggers building the documentation."""
-    command_list = [
+def _clear_previous_docs(output_directory: Path):
+    """Removes the existing output directory to avoid stale docs pages."""
+    if output_directory.is_dir():
+        shutil.rmtree(str(output_directory))
+
+
+def _generate_pdoc_command_list(output_directory: Path):
+    module = configuration.get_value(ConfigurationVariable.MODULE_TO_DOCUMENT)
```

Comment:
What I wasn't sure about with Path was is it worth taking `/some/folder` then converting it to a Path and then back to a string. I thought that that process would mean the path would then adapt to different filesystems and not be fixed like it is when a string.

Reply:
And then once I basically required a Path in most places it seemed more efficient to pass that around rather than spend a lot of time in each function converting it back and forth. But this is new to me so I'd like to know more.

Reply:
I agree with moving as much global state into main - that should be a fairly easy change because the chain of functions isn't too long right now.

Reply:
I'd prefer we passed `Path` objects around in our code and only converted them to `str`s when needed (like when passing to a `builtin`) this is so the type checker can help us more effectively, we make our intentions clearer to the reader and for consistency,

Reply:
Huh, I didn't consider all those things. 🤔 

To be honest, I don't know what running pdoc on windows would expect. `pdoc --output-dir C:\docs` or `pdoc --output-dir ./docs`? I suspect this is run in a bash-like environment anyway?

Now, about conversions:
- `_clear_previous_docs` - needs both string and `Path` representation of the path, so either as argument would be ok
- `_generate_pdoc_command_list` - needs string representation of the path (it currently takes `Path` and coverts it)
- `_generate_docs` - just forwards whatever it's given, doesn't care

In all honesty I'm not sure. My gut feeling would be to use string representations of paths and only `Path` them when necessary. I don't have strong feelings about this though.

Reply:
I am tempted to keep it with passing Path around at the moment, that feels better to me.